airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -810,8 +810,10 @@ class BigQueryGetDataOperator(GoogleCloudBaseOperator):
 
     :param dataset_id: The dataset ID of the requested table. (templated)
     :param table_id: The table ID of the requested table. (templated)
-    :param project_id: (Optional) The name of the project where the data
-        will be returned from. If None, it will be derived from the hook's project ID. (templated)
+    :param table_project_id: (Optional) The project ID of the requested table.
+        If None, it will be derived from the hook's project ID. (templated)
+    :param project_id: (Optional) Google Cloud Project where the job is running.
```

Comment:
You are changing what project_id means
If we want to do it can only be done in major release and even then its not something I am comfortable with as it may cause confusion for users.

My best advise here is deprecate project_id and use 2 new explicit parameter names.

Reply:
Thanks for the advice. I have reverted project_id's meaning and usage to as it was before. A new parameter job_project_id will handle job submission.